### DISCUSSION TITLE
consider to swap to mupdfjs ?

### DISCUSSION THREAD

**Discussion by panaC** [edrlab/thorium-reader#2742] [Developer corner]
It seems there is a new lib in the town :) [mupdf-js](https://github.com/ArtifexSoftware/mupdf.js/tree/master/examples/simple-viewer)

- wasm 
- c library well supported
- no need to web render pdf to get the cover img
- fast
- no viewer built-in, we have to create one specific to our needs

https://www.mupdf.com/

used in [sumatrapdf](https://www.sumatrapdfreader.org/free-pdf-reader)

**Comment by danielweck** [comment]
PDF.js may be a slower rendering engine, but it is a mature library, maintained / supported by many developers, with a strong feature set (including text selection + annotations) and a page layout logic that integrates quite well in Thorium.
I am not sure that the end-user benefits of a faster / lighter PDF rendering engine outweigh the risk of integration problems when adding WASM in Thorium's Electron stack. By the way, is the WASM library portable (i.e. identical binary) for all supported platforms: Linux, Windows, Mac in Intel and ARM target matrix?

**Comment by danielweck** [comment]
...perhaps a useful Proof Of Concept would be to reimplement the cover image extractor?
Minimum impact, easy to revert if we discover integration problems in one of our production builds for some target platforms?

**Comment by panaC** [comment]
> ...perhaps a useful Proof Of Concept would be to reimplement the cover image extractor? Minimum impact, easy to revert if we discover integration problems in one of our production builds for some target platforms?

This is my main target for the moment, indeed

**Comment by panaC** [comment]
We need to discuss the pros and cons, cost/benefit ratio of pdf-js and mupdf, I think.

At the moment, pdf-js with a large pdf file like Intel x86 manual (5082 pages) is extremely slow and barely usable with the chrome UI. Also the bottom bar (navigation) is broken (https://github.com/edrlab/thorium-reader/issues/1340).


https://github.com/edrlab/thorium-reader/issues/2670
